### PR TITLE
Extract file-type-icons icon mapping to JSON file to allow it to be consumed in non-JS environments

### PR DESCRIPTION
**Pull request by tachtevrenidis**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

As part of an effort to expose fabric file type icons to native mobile applications, we are proposing some changes, included in this PR.

This represents the first PR of the overall effort, which (1) factors out the file icon mapping into its own json file in order to be easier consumed by other environments (2) adds a script that will use that json file to produce an xcode compatible asset catalog.

#### Focus areas to test

All existing functionality is preserved, the only thing one could test is to ensure that the icon mapping is loaded correctly from the json file since it is not declared inline in the TS file anymore.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8633)

### CODE REVIEW

**Comment by tachtevrenidis** [comment]
@Jahnp per our chat earlier today...

File: scripts/generateAssetCatalog.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+const ICONS_FOLDER_PATH = argv.icons;
+const PATH_TO_SAVE_CATALOG = argv.out;
+
+if (!ICONS_FOLDER_PATH) {
+  throw new Error(
+    'Must specify the folder that contains the fabric-cdn icons. Use --icons=<icons> to provide a path (most often app-min/assets/item-types-fluent).'
+  );
+}
+
+if (!PATH_TO_SAVE_CATALOGn) {
```
Comment (KevinTCoughlin) [inline scripts/generateAssetCatalog.js]: typo?

Comment (tachtevrenidis) [reply scripts/generateAssetCatalog.js]: certainly... fixing...

**Comment by bigbadcapers** [comment]
@Jahnp can you check that there's no unforeseen side effects as a result of this PR? Kosta is adding support to auto-generate the required iOS package to always keep the filetype icons in sync across the platforms [File Coherence]

File: packages/file-type-icons/typings/import-json/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+declare module '*.json' {
```
Comment (ecraig12345) [inline packages/file-type-icons/typings/import-json/index.d.ts]: You shouldn't need this typing file. Let me know what error you're getting without it and I can suggest another strategy.

Comment (tachtevrenidis) [reply packages/file-type-icons/typings/import-json/index.d.ts]: importing json files in that way is only available on typescript > 2.9.x. This repo uses 2.8.x

File: packages/file-type-icons/tsconfig.json [hunk 3]
```diff
@@ -12,7 +12,10 @@
     "forceConsistentCasingInFileNames": true,
     "moduleResolution": "node",
     "preserveConstEnums": false,
-    "types": []
+    "typeRoots": ["node_modules/@types", "./typings"],
```
Comment (ecraig12345) [inline packages/file-type-icons/tsconfig.json]: Should be able to remove this now that you don't have the custom typing

Comment (tachtevrenidis) [reply packages/file-type-icons/tsconfig.json]: removed

**Comment by dzearing** [comment]
I am good with this. I pinged Katherine on Teams chat.

**Comment by tachtevrenidis** [comment]
thanks @dzearing

**Comment by tachtevrenidis** [comment]
> I am good with this. I pinged Katherine on Teams chat.

er... apparently I don't have permissions to merge PRs here... who can bless me with those?

**Comment by dzearing** [comment]
I can do it, but just following up with Katherine first; we'll get it done! :)

**Pull request merged**